ERCS/erc-7683.md

```diff
@@ -98,79 +118,241 @@ struct Input {
+	/// @dev Intended to improve standardized integration of various order types and settlement contracts
+	/// @param order The GaslessCrossChainOrder definition
+	/// @param originFillerData Any filler-defined data required by the settler
+	/// @returns ResolvedCrossChainOrder hydrated order data including the inputs and outputs of the order
+	function resolveFor(GaslessCrossChainOrder order, bytes originFillerData) external view returns (ResolvedCrossChainOrder);
 
-	/// @notice Resolves a specific CrossChainOrder into a generic ResolvedCrossChainOrder
+	/// @notice Resolves a specific OnchainCrossChainOrder into a generic ResolvedCrossChainOrder
 	/// @dev Intended to improve standardized integration of various order types and settlement contracts
-	/// @param order The CrossChainOrder definition
+	/// @param order The OnchainCrossChainOrder definition
 	/// @param fillerData Any filler-defined data required by the settler
 	/// @returns ResolvedCrossChainOrder hydrated order data including the inputs and outputs of the order
-	function resolve(CrossChainOrder order, bytes fillerData) external view returns (ResolvedCrossChainOrder);
+	function resolve(OnchainCrossChainOrder order) external view returns (ResolvedCrossChainOrder);
+}
+```
+
+A compliant destination settlement contract implementation MUST implement the `IDestinationSettler` interface:
+
+```solidity
+/// @title IDestinationSettler
+/// @notice Standard interface for settlement contracts on the destination chain
+interface IDestinationSettler {
+	/// @notice Fills a single leg of a particular order on the destination chain
+	/// @param orderId Unique order identifier for this order
+	/// @param originData Data emitted on the origin to parameterize the fill
+	/// @param fillerData Data provided by the filler to inform the fill or express their preferences
+	function fill(bytes32 orderId, bytes calldata originData, bytes calldata fillerData) external;
 }
 ```
 
+### fillerData
+
+Cross-chain execution systems implementing this standard SHOULD use a sub-type that can be parsed from the arbitrary `fillerData` field. This may include information such as the desired timing or form of payment for the filler
+
+All sub-types SHOULD be registered in the [Filler Data Subtypes Repo](github.com/erc-7683/filler-data-subtypes) to encourage sharing of sub-types based on their functionality.
+
 ## Rationale
 
 ### Generic OrderData
 
-A key consideration is to ensure that a broad range of cross-chain intent designs can work within the same standard. To enable this, the specification is designed around a standard cross-chain intents *flow*, while allowing for varying implementation details within that flow.
+A key consideration is to ensure that a broad range of cross-chain intent designs can work within the same standard. To enable this, the specification is designed around a cross-chain intents _flow_, with two variations: gasless and onchain.
```

Comment:
imo gasless and onchain might not be the best terms since you can have a gasless tx (eg using paymasters) that happens in the "onchain flow" and technically the "gasless flow" is no less onchain than the other. the only salient difference between these two flows is which order they are in so terminology that refers to the order might be easier to understand

Reply:
Do you have any alternative suggestions? I don't love the term onchain either.

Reply:
I think terms that mention the chronology make sense, here are some random ideas coming to mind (feel free to discard all of them but could be useful for brainstorming):
- pre-settle vs post-settle
- stepwise vs reverse
- stepwise vs instant
- pre-fund vs post-fund

Reply:
Would any of these be drop-in replacements for the two existing structs?

I'm having a little trouble mapping these chronology terms on the existing structs. Or are you suggesting functional changes?